### ISSUE TITLE
ParamType

### ISSUE THREAD

**Issue by gakuzzzz** [m3dev/octoparts#100]

**Comment by lloydmeta** [comment]
If I remember correctly, for body param types, we expect you to just send one + we don't it. I don't thnk this was made clear in the documentation though.

The reasoning was:
- There is, as far as I know, no "one way" for how to send a body to an HTTP endpoint.
- Thus, there are people POST'ing XML, JSON, message pack, EDN, you name it to their APIs

I'm not sure if have rethought the above, but I would suggest the following course of action:

1. At least for now, update docs to make it clear we don't handle more than 1 body parameters 
2. Think about whether or not we want to support validating body parameters
  - If so in what (limited) way ? How do we choose ?
  - Implement the feature
  - Update the docs again.

> We can not call HTTP Post API.

If you *can not call* the HTTP Post API, that is a different problem; it should still work. Maybe more details are needed.

**Comment by gakuzzzz** [comment]
Ok, I see. 
ParamType.Body is used any format as XML, JSON and so on.

I can not find the way how to setting `HTTP Post API` parameters.
Other ParamType s (Query, Path, Header, Cookie) does not seem usable to POST body.
How I do configure?

**Comment by mauhiz** [comment]
Right, it looks like currently your best shot is to prepare a post body client side and send it as a body parameter. This prevents you from validating parameters in Octoparts which is a bit of a waste ... I think this design miss was because we focused on REST.

**Comment by gakuzzzz** [comment]
I see. I understood that Octoparts behave as proxy so it bit difficult to parse and validate body.

I think, It is enough that Octoparts client and Admin console is possible to build body from some parameters. 
Octoparts server still does not parse and validate body.
And these parameters can not be used at cache control key. 

What do you think?

**Comment by lloydmeta** [comment]
I think that seems reasonable.

That said, to make this useful, we should make sure that the feature works across common body types; unless if there's a more scientific way, we should probably make sure the ones that [Play supports](https://github.com/playframework/playframework/blob/945e321887d3e5e550fe16e5c49b2784d0e3c7ca/framework/src/play/src/main/scala/play/api/http/Writeable.scala) can be built.  

Namely:
- XML
- JS
- Form
- normal string

**Comment by gakuzzzz** [comment]
I agree that the feature works across common body types.
